### ISSUE TITLE
Public key not available [Ubuntu 21.10]

### ISSUE THREAD

**Issue by repsorp** [meganz/MEGAsync#681]
```
W: GPG error: https://mega.nz/linux/repo/xUbuntu_21.10 ./ InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 1A664B787094A482
E: The repository 'https://mega.nz/linux/repo/xUbuntu_21.10 ./ InRelease' is not signed.
N: Updating from such a repository can't be done securely, and is therefore disabled by default.

```

Tried to add key with:
`sudo apt-key adv --keyserver keyserver.ubuntu.com --recv-keys 1A664B787094A482`
Didn't work, same error message.

**Comment by polmr** [comment]
MEGA repositories have a new strengthen key. This shall configure it:
```curl https://mega.nz/linux/repo/xUbuntu_21.10/Release.key |  sudo apt-key add -```

**Comment by repsorp** [comment]
> MEGA repositories have a new strengthen key. This shall configure it: `curl https://mega.nz/linux/repo/xUbuntu_21.10/Release.key | sudo apt-key add -`

Didn't work, got athe following warning message:
`Warning: apt-key is deprecated. Manage keyring files in trusted.gpg.d instead (see apt-key(8)).
`

**Comment by raniaamina** [comment]
you can try this one;

```bash
curl -s --compressed "https://mega.nz/linux/repo/xUbuntu_21.10/Release.key" | gpg --dearmor | sudo tee /usr/share/keyrings/mega-nz.gpg
```
then edit /etc/apt/sources.list.d/megasync.list
deb [signed-by=/usr/share/keyrings/mega-nz.gpg] https://mega.nz/linux/repo/xUbuntu_21.10/ ./

hope it help

**Comment by repsorp** [comment]
> hope it help

Still not working...
`/usr/share/keyrings/mega-nz.gpg` key file has been created but is empty

**Comment by repsorp** [comment]
Don't know why but it worked with the following command line:
`wget -O- https://mega.nz/linux/repo/xUbuntu_21.10/Release.key | gpg --dearmor | sudo tee /usr/share/keyrings/mega-nz.gpg`

Found it here:
https://www.linuxuprising.com/2021/01/apt-key-is-deprecated-how-to-add.html

**Comment by raniaamina** [comment]
> Thanks but still not working...

Ups sorry, i think it caused unused parameter in curl, my bad.
But you fixed it with wget, great!

**Comment by DracoThuban** [comment]
Check the URL in /etc/apt/source.list.  Was https://mega.nz/linux/repo/ but now is https://mega.nz/linux/MEGAsync/

**Comment by repsorp** [comment]
> Check the URL in /etc/apt/source.list. Was https://mega.nz/linux/repo/ but now is https://mega.nz/linux/MEGAsync/

Do not have any mega.nz URL in /etc/apt/source.list

**Comment by DracoThuban** [comment]
> > Check the URL in /etc/apt/source.list. Was https://mega.nz/linux/repo/ but now is https://mega.nz/linux/MEGAsync/
> 
> Do not have any mega.nz URL in /etc/apt/source.list

Sorry, I explained badly. Check the URL in the archive where you have repo's Mega.nz. I had the same problem. There are two folders with different dates.  https://mega.nz/linux/MEGAsync/ is the last.

**Comment by repsorp** [comment]
> Sorry, I explained badly. Check the URL in the archive where you have repo's Mega.nz. I had the same problem. There are two folders with different dates. https://mega.nz/linux/MEGAsync/ is the last.
 
Couldn't find these two folders with different dates...
Megasync repo 'megasync.list' is in `/etc/apt/sources.list`
Its content is  `deb https://mega.nz/linux/MEGAsync/xUbuntu_21.10/ ./`
I followed the instructions found in [Installing MEGA on Linux
](https://help.mega.io/installs-apps/desktop-syncing/linux) **How to install MEGA Desktop Application from our repositories**
`sudo echo "deb https://mega.nz/linux/MEGAsync/xUbuntu_21.04/ ./" > /etc/apt/sources.list.d/megasync.list`
I just changed 21.04 to 21.10
Seems to work fine so far...

**Comment by DracoThuban** [comment]
> 

This is the correct command:
`sudo echo "deb [signed-by=/usr/share/keyrings/mega-nz.gpg] https://mega.nz/linux/MEGAsync/xUbuntu_21.04/ ./" > /etc/apt/sources.list.d/megasync.list`

**Comment by DracoThuban** [comment]
And this for generate /usr/share/keyrings/mega-nz.gpg

`wget -O- https://mega.nz/linux/MEGAsync/xUbuntu_21.10/Release.key | gpg --dearmor | sudo tee /usr/share/keyrings/mega-nz.gpg`

**Comment by repsorp** [comment]
> > 
> 
> This is the correct command: `sudo echo "deb [signed-by=/usr/share/keyrings/mega-nz.gpg] https://mega.nz/linux/MEGAsync/xUbuntu_21.04/ ./" > /etc/apt/sources.list.d/megasync.list`

Correct command coming from where since it differs from the one in the megasync help doc?

**Comment by repsorp** [comment]
> And this for generate /usr/share/keyrings/mega-nz.gpg
> 
> `wget -O- https://mega.nz/linux/MEGAsync/xUbuntu_21.10/Release.key | gpg --dearmor | sudo tee /usr/share/keyrings/mega-nz.gpg`

Still doesn't work...
Same error message
```
W: An error occurred during the signature verification. The repository is not updated and the previous index files will be used. GPG error: https://mega.nz/linux/MEGAsync/xUbuntu_21.10 ./ InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 03C3AD3A7F068E5D
W: Failed to fetch https://mega.nz/linux/MEGAsync/xUbuntu_21.10/./InRelease  The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 03C3AD3A7F068E5D
```

**Comment by repsorp** [comment]
> > 
> 
> And this for generate /usr/share/keyrings/mega-nz.gpg
wget -O- https://mega.nz/linux/MEGAsync/xUbuntu_21.10/Release.key | gpg --dearmor | sudo tee /usr/share/keyrings/mega-nz.gpg

Nope, still doesn't work...

Following @polmr instructions correct command is :

Generate new strengthen key:
`curl -s "https://mega.nz/linux/repo/xUbuntu_21.10/Release.key"  | gpg --dearmor | sudo tee /usr/share/keyrings/mega-nz.gpg
`

Update `/etc/apt/source.list`:
`sudo echo "deb [signed-by=/usr/share/keyrings/mega-nz.gpg] https://mega.nz/linux/repo/xUbuntu_21.10/ ./" > /etc/apt/sources.list.d/megasync.list`

Now it's working

`sudo apt list --upgradable`
`megasync/unknown 4.6.5-3.1 amd64 [upgradable from: 4.6.3-6.1]`
